### ISSUE TITLE
Windowing arange on a rowless table causes panic

### ISSUE THREAD

**Issue by drhagen** [pola-rs/polars#2957]
#### What language are you using?

Python

#### What version of polars are you using?

0.13.14

#### What operating system are you using polars on?

Linux

#### What language version are you using

Python 3.10.2

#### Describe your bug.

It appears there is a corner case where these three things are applied all at once in a `with_column`:

1. `arange` to make a vector that is the correct length
2. windowed using `over`
3. rowless table

If anyone is wondering why I would want to do this, I am generating the within-group row indexes. But is crashes when the table has no rows.

#### What are the steps to reproduce the behavior?

```python
import polars as pl

df = pl.DataFrame(dict(x=[]))

df.with_column(pl.arange(0,pl.count()).over('x'))
# pyo3_runtime.PanicException: assertion failed: index < self.len()
```

#### What is the expected behavior?

```
shape: (0, 2)
┌─────┬─────────┐
│ x   ┆ literal │
│ --- ┆ ---     │
│ f32 ┆ i64     │
╞═════╪═════════╡
└─────┴─────────┘
```

**Comment by drhagen** [comment]
Actually, now that I play with this more, it looks like `arange` is unsafe when windowed. If the table is not rowless, then I get non-deterministic values in the resulting column.

```python
import polars as pl

df = pl.DataFrame(dict(x=[5,5,4,4,2,2]))

df.with_column(pl.arange(0,pl.count()).over('x'))
shape: (6, 2)
# ┌─────┬───────────────┐
# │ x   ┆ arange        │
# │ --- ┆ ---           │
# │ i64 ┆ i64           │
# ╞═════╪═══════════════╡
# │ 5   ┆ 1             │
# ├╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ 5   ┆ 1             │
# ├╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ 4   ┆ 0             │
# ├╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ 4   ┆ 0             │
# ├╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ 2   ┆ 6507786799376 │
# ├╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
# │ 2   ┆ 6507786799376 │
# └─────┴───────────────┘

```

**Comment by ritchie46** [comment]
Thanks, I will take a look.

**Comment by ritchie46** [comment]
fixed by #2958
